Lesson6

vplrt opened the pull request:
[#116769807] Add comments to lessons and homeworks.
[#116769765] Fix image uploading. File will not disapear when validation error.
[#116769841] Coach can reject homework only if commented.

![rubocop_l6](https://cloud.githubusercontent.com/assets/6698143/14353476/45ff3c50-fcea-11e5-9e72-b25cfb79225c.png)

Commit:
[#116769807] Add comments to lessons and homeworks.

Commit:
[#116769765] Fix image uploading. File will not disapear when validation error.

vplrt commented on app/controllers/users/homeworks_controller.rb:
```diff
@@ -16,11 +16,18 @@ def create
   def reject
-    @homework.reject!
+    authorize! :reject, Homework
+    if @homework.comments.where(user_id: current_user.id).any?
```
Не стал добавлять сервис для этого экшна т.к. логика простая.
Возможно @homework.comments.where(user_id: current_user.id).any? нужно было вынести в отдельный метод.

c3gdlk replied:
так логика неправильная. Допустим я уже комментировал домашнее задание. При реджекте я должен указать причину.

Кстати exists? отработает быстрее

c3gdlk commented on app/controllers/users/comments_controller.rb:
```diff
@@ -0,0 +1,40 @@
+class Users::CommentsController < ApplicationController
```
а почему контроллер в неймспейе users? Неймспейс users - это личный кабинет

c3gdlk commented on app/models/ability.rb:
```diff
@@ -24,10 +25,14 @@ def user_abilities(user)
     can :manage, [Profile], user_id: user.id
     can :create, Homework
-    cannot [:read, :approve, :reject], Homework
+    can :read, Homework, user_id: user.id
```
а где роли в файле abilities?

vplrt replied:
```
  def initialize(user)
    if user.present?
      user.has_role?(:coach) ? coach_abilities(user) : user_abilities(user)
    else
      guest_abilities
    end
  end
```

c3gdlk replied:
точно? гит спрятал

Commit:
[#116769841] Coach can reject homework only with comment.
[#116769807] Delete Users namespace for CommentsController. Fix validation errors in comments form.

c3gdlk commented on app/controllers/application_controller.rb:
```diff
@@ -5,7 +5,12 @@ class ApplicationController < ActionController::Base
   layout :layout_by_resource
 
-  rescue_from CanCan::AccessDenied, with: :not_autorized
+  rescue_from CanCan::AccessDenied do
```
блоками обычно не делают. Лучше все  таки метод. Потому что на метод можно написать тест

c3gdlk commented:
👍 когда поправите последний

Commit:
Fix exceptions in application_controller
